allensdk/brain_observatory/filtered_running_speed/__main__.py

```diff
@@ -0,0 +1,954 @@
+    """
+    # Compare against previous value
+    shifted_vsig = _shift(vsig)
+
+    if not isinstance(vsig, np.ndarray):
+        vsig = np.array(vsig)
+
+    # Suppress warnings for when comparing to nan values
+    with np.errstate(invalid='ignore'):
+        pos_wraps = np.asarray(
+            np.logical_and(vsig < min_threshold, shifted_vsig > max_threshold)
+            ).nonzero()[0]
+        neg_wraps = np.asarray(
+            np.logical_and(vsig > max_threshold, shifted_vsig < min_threshold)
+            ).nonzero()[0]
+
+    return pos_wraps, neg_wraps
+
+
+def extract_running_speeds(
+    dx_raw,
+    frame_times,
+    vsig,
+    vin,
+    start_index,
+    end_index,
+    lowpass: bool = True,
+    zscore_threshold=DEFAULT_ZSCORE_THRESHOLD,
+):
+    """
+    Given the dx_deg from the 'pkl' file object and a 1d
+    array of timestamps, compute the running speed. Returns a
+    dataframe with the raw voltage dx_deg as well as the computed speed
+    at each timestamp. By default, the running speed is filtered with
+    a 10 Hz Butterworth lowpass filter to remove artifacts caused by
+    the rotary encoder.
+
+    Parameters
+    ----------
+    dx_raw
+        Deserialized 'pkl' file dx_deg
+    frame_times: np.ndarray (1d)
+        Timestamps for running data measurements
+    vsig: np.ndarray (1d),
+        raw analog data for encoder
+    vin: np.ndarray (1d),
+        input voltage to the encoder
+    start_index: int,
+        the start index of the stimulus in reference to the full vsync times
+    end_index: int,
+        the end index of the stimulus in reference to the full vsync times
+    lowpass: bool (default=True)
+        Whether to apply a 10Hz low-pass filter to the running speed
+        data.
+    zscore_threshold: float
+        The threshold to use for removing outlier running speeds which might
+        be noise and not true signal.
+
+    Returns
+    -------
+    pd.DataFrame
+        Dataframe with an index of timestamps and the following
+        columns:
+            "velocity": computed running speed
+            "net_rotation": dx in radians
+            "frame_indexes": frame indexes into the raw frame time,
+            "frame_times": the frame times for the timestamp data
+
+    Notes
+    -----
+    Though the angular change is available in the raw data
+    (key="dx"), this method recomputes the angular change from the
+    voltage signal (key="vsig") due to very specific, low-level
+    artifacts in the data caused by the encoder. See method
+    docstrings for more detailed information. The raw data is
+    included in the final output in case the end user wants to apply
+    their own corrections and compute running speed from the raw
+    source.
+    """
+
+    # Identify "wraps" in the voltage signal that need to be unwrapped
+    # This is where the encoder switches from 0V to 5V or vice versa
+    pos_wraps, neg_wraps = _identify_wraps(
+        vsig, min_threshold=1.5, max_threshold=3.5)
```

Comment:
should `min_threshold` and `max_threshold` be parameters exposed in this method's call signature, in case they ever change (or need to be set to different values on different rigs)?

Reply:
This has been removed from my code as I am now calling a library that contains this.